src/julia_gc.c

```diff
@@ -474,6 +474,20 @@ static void TryMark(void * p)
     }
 }
 
+static void TryMarkRangeReverse(void * start, void * end)
+{
+    if (lt_ptr(end, start)) {
+        SWAP(void *, start, end);
+    }
+    char * p = (char *)(align_ptr(start));
+    char * q = (char *)end - sizeof(void *);
+    while (!lt_ptr(q, p)) {
+        void *addr = *(void **)q;
+        if (addr) TryMark(addr);
```

Comment:
If `TryMark` require non-null arguments, then perhaps that check should be done in `TryMark`? Perhaps `TryMark` also should be marked as `inline`?

Reply:
It doesn't require non-null arguments, this is simply an optimization.